python/cuml/internals/input_utils.py

```diff
@@ -163,6 +163,14 @@ def get_supported_input_type(X):
     if isinstance(X, CudfDataFrame):
         return CudfDataFrame
 
+    # A cudf.pandas wrapped Numpy array defines `__cuda_array_interface__`
+    # which means without this we'd always return a cupy array. We don't want
+    # to match wrapped cupy arrays, they get dealt with later
+    if hasattr(X, "_fsproxy_fast_type") and X.__class__.__module__.endswith(
+        ".numpy"
+    ):
+        return np.ndarray
```

Comment:
```suggestion
    if (np_array := getattr(arr, "_fsproxy_slow_type", None)) is np.ndarray:
        return np_array
```

Yeah not a great way to check for this today :/

Reply:
@mroeschke is this something prone to break when cudf.pandas changes anytime soon, or would this check be good for the time being?

Reply:
This attribute is used pretty heavily internally in `cudf.pandas` so I don't anticipate this breaking anytime soon and good for the time being.

Reply:
I like it. I used a variation on your suggestion which I think is equivalent and I find easier to understand (using the result of `getattr` makes me think "why use this? Is it somehow different from `np.ndarray`? But why call it `np_array` then? Is it a `np.array`? - maybe I should think less while reading code?)